src/fluid.rs

```diff
@@ -0,0 +1,434 @@
+        self.density[idx] += amount;
+        self.s[idx] += amount;
+    }
+
+    fn add_velocity(&mut self, x: u32, y: u32, amount_x: f32, amount_y: f32) {
+        let idx: usize = Fluid::coordinate_to_idx(x, y, &self.simulation_configs.size);
+        self.velocities_x[idx] += amount_x;
+        self.velocities_y[idx] += amount_y;
+    }
+
+    fn render_density(&self, imgbuf: &mut image::RgbImage, frame_number: u32) {
+        for (x, y, pixel) in imgbuf.enumerate_pixels_mut() {
+            let density =
+                self.density[Fluid::coordinate_to_idx(x, y, &self.simulation_configs.size)];
+            *pixel = image::Rgb([(density * 255.0) as u8, 200, density as u8]);
+        }
+        let img_name = format!("rendered_images/density{}.jpg", frame_number);
+        imgbuf.save(img_name).unwrap();
+    }
+
+    fn set_boundaries(b: u32, x: &mut [f32], size: &u32) {
+        for i in 1..size - 1 {
+            x[Fluid::coordinate_to_idx(i, 0, size)] = if b == 2 {
+                -x[Fluid::coordinate_to_idx(i, 1, size)]
+            } else {
+                x[Fluid::coordinate_to_idx(i, 1, size)]
+            };
+            x[Fluid::coordinate_to_idx(i, size - 1, size)] = if b == 2 {
+                -x[Fluid::coordinate_to_idx(i, size - 2, size)]
+            } else {
+                x[Fluid::coordinate_to_idx(i, size - 2, size)]
+            };
+        }
+
+        for j in 1..size - 1 {
+            x[Fluid::coordinate_to_idx(0, j, size)] = if b == 1 {
+                -x[Fluid::coordinate_to_idx(1, j, size)]
+            } else {
+                x[Fluid::coordinate_to_idx(1, j, size)]
+            };
+            x[Fluid::coordinate_to_idx(size - 1, j, size)] = if b == 1 {
+                -x[Fluid::coordinate_to_idx(size - 2, j, size)]
+            } else {
+                x[Fluid::coordinate_to_idx(size - 2, j, size)]
+            };
+        }
+
+        x[Fluid::coordinate_to_idx(0, 0, size)] = 0.5
+            * (x[Fluid::coordinate_to_idx(1, 0, size)] + x[Fluid::coordinate_to_idx(0, 1, size)]);
+        x[Fluid::coordinate_to_idx(0, size - 1, size)] = 0.5
+            * (x[Fluid::coordinate_to_idx(1, *size - 1, size)]
+                + x[Fluid::coordinate_to_idx(0, *size - 2, size)]);
+        x[Fluid::coordinate_to_idx(size - 1, 0, size)] = 0.5
+            * (x[Fluid::coordinate_to_idx(size - 2, 0, size)]
+                + x[Fluid::coordinate_to_idx(size - 1, 1, size)]);
+        x[Fluid::coordinate_to_idx(size - 1, size - 1, size)] = 0.5
+            * (x[Fluid::coordinate_to_idx(size - 2, size - 1, size)]
+                + x[Fluid::coordinate_to_idx(size - 1, size - 2, size)]);
+    }
+
+    fn diffuse(
+        b: u32,
+        x: &mut [f32],
+        x0: &[f32],
+        diffusion: &f32,
+        size: &u32,
+        dt: &f32,
+        iter: &u32,
+    ) {
+        let size_float: f32 = (size - 2) as f32;
+        let a: f32 = dt * diffusion * size_float * size_float;
```

Comment:
no need to specify that the values are f32 when declaring them

Reply:
It is just uneccessary and not very consistent, in some places you don't specify the type and on others you do. the line on 161 is you have already explicitly added the type with `as f32`. You decide how to leave it

Reply:
OK, I'm removing the explicit declaration of the type